src/MPoly.jl

```diff
@@ -416,13 +416,14 @@ end
 iszero(x::MPolyRingElem{T}) where T <: RingElement = length(x) == 0
 
 function is_unit(f::T) where {T <: MPolyRingElem}
-  # for constant polynomials we delegate to the coefficient ring:
-  is_constant(f) && return is_unit(constant_coefficient(f))
-  # Here deg(f) > 0; over an integral domain, non-constant polynomials are never units:
-  is_domain_type(T) && return false
   # A polynomial over a commutative ring is a unit iff its
   # constant term is a unit and all other coefficients are nilpotent:
   # see e.g. <https://kconrad.math.uconn.edu/blurbs/ringtheory/polynomial-properties.pdf> for a proof.
+  # For constant polynomials we delegate to the coefficient ring:
+  is_constant(f) && return is_unit(constant_coefficient(f))
+  # Here deg(f) > 0.
+  # Over an integral domain, non-constant polynomials are never units:
+  is_domain_type(T) && return false
```

Comment:
```suggestion
  is_domain_type(base_ring_type(T)) && return false
```
It should be the same over commutative rings, but this fits the comment better I believe